packages/common/src/ide/types/RecordedTestStorage.ts

```diff
@@ -0,0 +1,11 @@
+import type { RecordTestCaseCommandOptions } from "../../testUtil/RecordTestCaseCommandOptions";
+
+export interface TestCaseRecorderStorage {
```

Comment:
This is the one provider I'm not quite happy with. It's to close to the file system and had don't abstract enough. The problem is that its a lot of path logic in the test case recorder and we basically would be moving all of that into a environment specific implementation. We need to have a discussion about this.

Reply:
I agree this is awkward. I thought we decided test case recorder doesn't make sense without file system so we might as well just put it in its own package and let it use the file system as it wants

We had actually discussed at some point that test case recorder might someday be used independent of cursorless, so moving it out of the engine seems reasonable

Reply:
Yeah that is probably where I want to go in the long run. This pull request just got so large that I'm trying to keep it down, but might be worth doing here anyhow?

Reply:
Ok tests case recorder is now moved into a separate package :)